src/stores/FarcasterStore.ts

```diff
@@ -6,29 +6,35 @@ import postIdsStatuses from 'stores/PostIdsStatuses'
 const farcasterStore = proxy<{
   casts: { [merkleRoot: string]: Promise<Cast> }
   threads: { [postId: number]: Promise<string[]> }
+  requested: { [postId: number]: boolean }
 }>({
   casts: {},
   threads: {},
+  requested: {},
 })
 
-export function fetchFarcasterThread(postId: number) {
-  if (typeof farcasterStore.threads[postId] !== 'undefined') return
-  const request = fetchThreadByPostId(postId)
-  farcasterStore.threads[postId] = request.then(({ casts }) =>
-    casts.map((cast) => cast.merkleRoot)
-  )
-  void request.then(({ casts, merkleRoot }) => {
+export async function fetchFarcasterThread(postId: number) {
+  if (farcasterStore.requested[postId]) return
+  farcasterStore.requested[postId] = true
+
+  try {
+    const { casts, merkleRoot } = await fetchThreadByPostId(postId)
+    const merkles = casts.map((cast) => cast.merkleRoot)
+
+    farcasterStore.threads[postId] = Promise.resolve(merkles)
     if (merkleRoot)
       postIdsStatuses.idToMerkleRoot[postId] = Promise.resolve(merkleRoot)
-    for (const cast of casts) {
+    casts.forEach((cast) => {
       farcasterStore.casts[cast.merkleRoot] = Promise.resolve(cast)
       if (cast.postId) {
```

Comment:
Let's remove curlies

Reply:
The next function call takes three lines, and in order not to confuse the developer, I think it's better to leave the curly brace